src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,117 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertManufacturerQuery = "INSERT INTO manufacturers(name, country) VALUES(?,?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement createManufacturerStatement = connection
+                           .prepareStatement(insertManufacturerQuery,
+                                 Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setString(1, manufacturer.getName());
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet generatedKeys = createManufacturerStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert manufacturer to DB " + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getManufacturerByIdQuery = "SELECT * FROM manufacturers WHERE id = ? "
+                + "AND is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                     PreparedStatement getAllManufacturers = connection
+                        .prepareStatement(getManufacturerByIdQuery)) {
+            getAllManufacturers.setObject(1, id);
+            ResultSet resultSet = getAllManufacturers
+                    .executeQuery();
+            return Optional.of(new ManufacturerDaoImpl().extractManufacturer(resultSet));
```

Comment:
```suggestion
            return Optional.of(extractManufacturer(resultSet));
```
why do you need to create dao instance here?

Reply:
> why do you need to create dao instance here?

fixed)